### DISCUSSION TITLE
Missing CLI functions & missing web browser debug

### DISCUSSION THREAD

**Discussion by elementalTIMING** [elgatosf/elgato#107] [Help]
Hi,

I tried the new CLI tool and it's really nice so far. But how can I get access to the described features, like 'pack' and 'validate'? It's described in the README on GitHub but the features are missing in the package. E.g.

	streamdeck pack
	error: unknown command 'pack'

Further and more important: how to debug in the browser? The plugin is not shown when accessing `http://localhost:23654/`. This worked in the "legacy style", but how to get t to work now? The problem is: no browser debug -> no plugin debug -> no new plugin.

**UPDATE**
meanwhile I found out that the browser inspector just works, when a in the Elgato Stream Deck desktop software is selected BEFORE opening the url `http://localhost:23654`. A bit strange to work with!

**NEW Question**
is there any example for the *Property Inspector* working with the new TypeScript plugin template? 

Please help! Thx!

**Comment by GeekyEggo** [comment]
Hey Lars,

## `pack` command

The `pack` and `validate` commands are currently released under the `@beta` pre-release identifier; to access these, you can install the beta version via npm:
```
npm install -g @elgato/cli@beta
```

## Debugging

Debugging Node.js plugins is configurable via the new `Nodejs` manifest option ([read more](https://github.com/elgatosf/streamdeck/?tab=readme-ov-file#-debugging)), specifically the `Nodejs.Debug` value. 

One significant advantage of Node.js plugins is that you can now debug them from within your IDE! With the plugin running, in VS Code:
1. Open the command palette (`Ctrl` + `P` on Windows).
2. Enter `> Debug: Attach to Node Process`.
3. Select the plugin from the dropdown (the process name will be `node20`.

Alternatively, debugging from within the browser is still possible via listening to the port number. More information on debugging Node.js plugins in the browser can be found on Node.js' [Debugging Node.js](https://nodejs.org/en/learn/getting-started/debugging) page.

## Property Inspector

The property inspector is currently in development for the new SDK; we'll be sharing more in the coming weeks, so keep a eye on our [Makers Discord](https://discord.gg/GehBUcu627) for the latest announcements.
